sklearn/linear_model/base.py

```diff
@@ -90,8 +90,15 @@ def _preprocess_data(X, y, fit_intercept, normalize=False, copy=True,
     if isinstance(sample_weight, numbers.Number):
         sample_weight = None
 
-    X = check_array(X, copy=copy, accept_sparse=['csr', 'csc'],
-                    dtype=FLOAT_DTYPES)
+    if check_input:
+        X = check_array(X, copy=copy, accept_sparse=['csr', 'csc'],
+                        dtype=FLOAT_DTYPES)
+    elif copy:
+        if sp.issparse(X):
+            X = X.copy()
```

Comment:
This isn't covered by tests.

Reply:
It seems there are simply not tests for when `X` is sparse. `copy=True` is the default and none of the tests in `sklearn/linear_model/tests/test_base.py` seem to set `copy` otherwise. How would you like me to proceed?

Reply:
Ok, think I've fixed this.